src/store/user/userSlice.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import { httpService, _apiBase } from '../../services';
+import { IFulfilledLoginUser } from '../../types';
+
+const { request } = httpService();
+
+export const fetchLoginUser = createAsyncThunk(
+	'user/fetchLoginUser',
+	(user: string) => {
+		return request(`${_apiBase}login`, 'POST', user);
+	}
+);
+
+export const fetchLogoutUser = createAsyncThunk(
+	'user/fetchLogoutUser',
+	(user: string) => {
+		return request(`${_apiBase}delete`, 'DELETE', user);
+	}
+);
+
+interface IInitialState {
+	isAuth: boolean;
+	name: string;
+	email: string;
+	token: string;
+	role: string;
+}
+
+const initialState: IInitialState = {
+	isAuth: false,
+	name: '',
+	email: '',
+	token: '',
+	role: '',
+};
+
+const userSlice = createSlice({
+	name: 'user',
+	initialState,
+	reducers: {
+		onLogoutClickClearState: (state) => initialState,
```

Comment:
I'd typically say that action naming should be independent from their implementation
this is to avoid confusion and unnecessary renaming, for example if on logout click you also need to call a /logout api, you'd have to rename the action. avoiding implementation details in naming, and using something like `onLogout` or `onLogoutClick` would decouple and get you more flexibility

or as redux states much better than me: https://redux.js.org/style-guide/

However, we recommend trying to treat actions more as "describing events that occurred", rather than "setters". Treating actions as "events" generally leads to more meaningful action names, fewer total actions being dispatched, and a more meaningful action log history. Writing "setters" often results in too many individual action types, too many dispatches, and an action log that is less meaningful.